Thanks for the PR @aayushprsingh!

 There's a TypeScript compilation error in src/pages/InterviewPrepPage.tsx:
error TS2686: 'React' refers to a UMD global, but the current file is a module.
Please fix by replacing <React.Fragment> with shorthand empty tags <> and </> on lines 402 and 462.

Let me know once fixed!

--- a/backend/app/main.py
+++ b/backend/app/main.py
@@ -378,6 +378,7 @@ class GapItem(BaseModel):
     have: str
     need: str
     gapLevel: Literal["Low", "Medium", "High"]
+    resources: list[str] = Field(default_factory=list)
 
 
 class QuestionItem(BaseModel):
@@ -755,11 +756,12 @@ async def generate_session_payload(
                 "You generate structured interview preparation data. "
                 "Return valid JSON only. Do not include markdown or explanations. "
                 "Use exactly this schema: "
-                '{"gapAnalysis":[{"skill":"string","have":"string","need":"string","gapLevel":"Low|Medium|High"}],'
+                '{"gapAnalysis":[{"skill":"string","have":"string","need":"string","gapLevel":"Low|Medium|High","resources":["string"]}],'
                 '"readinessScore":0,'
                 '"questionBank":[{"question":"string","type":"behavioral|technical|situational","difficulty":"easy|medium|hard","tip":"string"}],'
                 '"roadmap":[{"day":1,"focusArea":"string","tasks":["string"]}]}. '
-                "gapAnalysis must be an array with 3 to 5 items. "
+                "gapAnalysis must be an array with 3 to 5 items. For each gapAnalysis item, "
+                "include 2-3 resource keywords (e.g., 'React', 'MDN', 'System Design', 'Python Docs'). "
                 "questionBank must be an array with 6 to 10 items. "
                 "roadmap must be an array with exactly 5 days."
             ),
@@ -797,6 +799,7 @@ async def generate_session_payload(
             "have": ["have", "current"],
             "need": ["need", "required"],
             "gapLevel": ["gaplevel", "level", "gap"],
+            "resources": ["resources", "links", "keywords"],
         }
         q_mapping = {
             "question": ["question", "text"],
@@ -811,11 +814,13 @@ async def generate_session_payload(
         }
 
         raw_gap = norm_res.get("gapanalysis", [])
-        gap_analysis = [
-            GapItem(**norm_dict(item, gap_mapping))
-            for item in raw_gap
-            if isinstance(item, dict)
-        ]
+        gap_analysis = []
+        for item in raw_gap:
+            if isinstance(item, dict):
+                normed = norm_dict(item, gap_mapping)
+                if normed.get("resources") is None:
+                    normed["resources"] = []
+                gap_analysis.append(GapItem(**normed))
 
         readiness_val = norm_res.get("readinessscore", norm_res.get("readiness", 50))
         readiness = max(0, min(100, int(readiness_val)))
@@ -844,12 +849,40 @@ async def generate_session_payload(
     scores = compute_match_score(resume_text, jd_text)
     readiness = scores["overallScore"]
     gap_analysis = [
-        GapItem(skill="React", have="Intermediate", need="Advanced", gapLevel="Medium"),
-        GapItem(skill="System Design", have="Basic", need="Advanced", gapLevel="High"),
-        GapItem(skill="TypeScript", have="Advanced", need="Advanced", gapLevel="Low"),
-        GapItem(skill="CI/CD", have="Basic", need="Intermediate", gapLevel="Medium"),
         GapItem(
-            skill="Testing", have="Intermediate", need="Advanced", gapLevel="Medium"
+            skill="React",
+            have="Intermediate",
+            need="Advanced",
+            gapLevel="Medium",
+            resources=["React", "MDN"],
+        ),
+        GapItem(
+            skill="System Design",
+            have="Basic",
+            need="Advanced",
+            gapLevel="High",
+            resources=["System Design", "MDN"],
+        ),
+        GapItem(
+            skill="TypeScript",
+            have="Advanced",
+            need="Advanced",
+            gapLevel="Low",
+            resources=["TypeScript", "MDN"],
+        ),
+        GapItem(
+            skill="CI/CD",
+            have="Basic",
+            need="Intermediate",
+            gapLevel="Medium",
+            resources=["CI/CD", "Docker"],
+        ),
+        GapItem(
+            skill="Testing",
+            have="Intermediate",
+            need="Advanced",
+            gapLevel="Medium",
+            resources=["Testing", "MDN"],
         ),
     ]
     question_bank = [

--- a/src/lib/store.ts
+++ b/src/lib/store.ts
@@ -70,6 +70,7 @@ export interface GapItem {
   have: string;
   need: string;
   gapLevel: "Low" | "Medium" | "High";
+  resources: string[];
 }
 
 export interface QuestionItem {

--- a/src/pages/CareerDNAPage.tsx
+++ b/src/pages/CareerDNAPage.tsx
@@ -79,13 +79,17 @@ export default function CareerDNAPage({ user, profile }: CareerDNAPageProps) {
 
         <Section title="Work History">
           <div className="space-y-3">
-            {profile.workHistory.filter((w) => w.jobTitle).map((w) => (
-              <div key={w.id} className="p-3 rounded-lg bg-secondary/30">
-                <p className="text-sm font-medium text-foreground">{w.jobTitle} at {w.company}</p>
-                <p className="text-xs text-muted-foreground">{w.from} — {w.to}</p>
-                {w.responsibilities && <p className="text-xs text-muted-foreground mt-1">{w.responsibilities}</p>}
-              </div>
-            ))}
+            {profile.workHistory.filter((w) => w.jobTitle).length === 0 ? (
+              <p className="text-sm text-muted-foreground italic">No work experience listed (Fresher)</p>
+            ) : (
+              profile.workHistory.filter((w) => w.jobTitle).map((w) => (
+                <div key={w.id} className="p-3 rounded-lg bg-secondary/30">
+                  <p className="text-sm font-medium text-foreground">{w.jobTitle} at {w.company}</p>
+                  <p className="text-xs text-muted-foreground">{w.from} — {w.to}</p>
+                  {w.responsibilities && <p className="text-xs text-muted-foreground mt-1">{w.responsibilities}</p>}
+                </div>
+              ))
+            )}
           </div>
         </Section>
 

--- a/src/pages/InterviewPrepPage.tsx
+++ b/src/pages/InterviewPrepPage.tsx
@@ -1,4 +1,4 @@
-import { useState, useRef, useEffect } from "react";
+import { useState, useRef, useEffect, Fragment } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
 import { motion } from "framer-motion";
 import { BookOpen, Loader2, Search, Brain, Cpu, Upload, Trash2, ChevronDown } from "lucide-react";
@@ -431,7 +431,7 @@ export default function InterviewPrepPage({
                       {activeSession.gapAnalysis.map((g, idx) => {
                         const isExpanded = !!expandedGaps[idx];
                         return (
-                          <React.Fragment key={idx}>
+                          <Fragment key={idx}>
                             <tr
                               onClick={() => toggleGap(idx)}
                               className="border-b border-border/50 hover:bg-secondary/20 transition-colors group cursor-pointer"
@@ -491,7 +491,7 @@ export default function InterviewPrepPage({
                                 </td>
                               </tr>
                             )}
-                          </React.Fragment>
+                          </Fragment>
                         );
                       })}
                     </tbody>

--- a/src/pages/InterviewPrepPage.tsx
+++ b/src/pages/InterviewPrepPage.tsx
@@ -1,7 +1,7 @@
 import { useState, useRef, useEffect } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
 import { motion } from "framer-motion";
-import { BookOpen, Loader2, Search, Brain, Cpu, Upload, Trash2 } from "lucide-react";
+import { BookOpen, Loader2, Search, Brain, Cpu, Upload, Trash2, ChevronDown } from "lucide-react";
 import { Button } from "@/components/ui/button";
 import {
   AlertDialog,
@@ -182,6 +182,50 @@ export default function InterviewPrepPage({
     Low: "text-success",
     Medium: "text-warning",
     High: "text-destructive",
+  };
+
+  const RESOURCE_MAPPING: Record<string, { label: string; url: string }> = {
+    React: { label: "React Docs", url: "https://react.dev/" },
+    "React.js": { label: "React Docs", url: "https://react.dev/" },
+    Python: { label: "Python Docs", url: "https://docs.python.org/3/" },
+    "Python Docs": { label: "Python Docs", url: "https://docs.python.org/3/" },
+    SQL: { label: "SQL MDN", url: "https://developer.mozilla.org/en-US/docs/Glossary/SQL" },
+    JavaScript: { label: "MDN JS", url: "https://developer.mozilla.org/en-US/docs/Web/JavaScript" },
+    MDN: { label: "MDN Web Docs", url: "https://developer.mozilla.org/en-US/" },
+    "System Design": { label: "System Design Primer", url: "https://github.com/donnemartin/system-design-primer" },
+    TypeScript: { label: "TypeScript Docs", url: "https://www.typescriptlang.org/docs/" },
+    Docker: { label: "Docker Docs", url: "https://docs.docker.com/" },
+    Kubernetes: { label: "Kubernetes Docs", url: "https://kubernetes.io/docs/home/" },
+    AWS: { label: "AWS Docs", url: "https://docs.aws.amazon.com/" },
+    "Node.js": { label: "Node.js Docs", url: "https://nodejs.org/docs/" },
+    FastAPI: { label: "FastAPI Docs", url: "https://fastapi.tiangolo.com/" },
+    Go: { label: "Go Docs", url: "https://go.dev/doc/" },
+    Java: { label: "Java Docs", url: "https://docs.oracle.com/en/java/" },
+    "C++": { label: "C++ Reference", url: "https://en.cppreference.com/w/" },
+    "C#": { label: "C# Docs", url: "https://learn.microsoft.com/en-us/dotnet/csharp/" },
+    Spring: { label: "Spring Docs", url: "https://spring.io/projects/spring-framework" },
+    Django: { label: "Django Docs", url: "https://docs.djangoproject.com/" },
+    Flask: { label: "Flask Docs", url: "https://flask.palletsprojects.com/" },
+    Redux: { label: "Redux Docs", url: "https://redux.js.org/" },
+    GraphQL: { label: "GraphQL Docs", url: "https://graphql.org/learn/" },
+    Tailwind: { label: "Tailwind Docs", url: "https://tailwindcss.com/docs" },
+    CSS: { label: "MDN CSS", url: "https://developer.mozilla.org/en-US/docs/Web/CSS" },
+    HTML: { label: "MDN HTML", url: "https://developer.mozilla.org/en-US/docs/Web/HTML" },
+    Testing: { label: "Testing Library", url: "https://testing-library.com/docs/" },
+    "CI/CD": { label: "CI/CD Guide", url: "https://resources.github.com/ci-cd/" },
+    Git: { label: "Git Docs", url: "https://git-scm.com/doc" },
+    PostgreSQL: { label: "PostgreSQL Docs", url: "https://www.postgresql.org/docs/" },
+    MongoDB: { label: "MongoDB Docs", url: "https://www.mongodb.com/docs/" },
+    Redis: { label: "Redis Docs", url: "https://redis.io/docs/" },
+  };
+
+  const [expandedGaps, setExpandedGaps] = useState<Record<number, boolean>>({});
+
+  const toggleGap = (idx: number) => {
+    setExpandedGaps((prev) => ({
+      ...prev,
+      [idx]: !prev[idx],
+    }));
   };
 
   const filteredQuestions = activeSession?.questionBank.filter((q) => {
@@ -372,7 +416,7 @@ export default function InterviewPrepPage({
                   <table className="w-full text-sm">
                     <thead>
                       <tr className="border-b border-border">
-                        <th className="text-left py-3 text-muted-foreground font-medium">Skill</th>
+                        <th className="text-left py-3 text-muted-foreground font-medium pl-4">Skill</th>
                         <th className="text-left py-3 text-muted-foreground font-medium">
                           {activeSession.isEstimated && !activeSession.resumeText && !activeSession.jdText
                             ? "You Have (Estimated)"
@@ -380,17 +424,76 @@ export default function InterviewPrepPage({
                         </th>
                         <th className="text-left py-3 text-muted-foreground font-medium">They Need</th>
                         <th className="text-left py-3 text-muted-foreground font-medium">Gap Level</th>
+                        <th className="w-10"></th>
                       </tr>
                     </thead>
                     <tbody>
-                      {activeSession.gapAnalysis.map((g) => (
-                        <tr key={g.skill} className="border-b border-border/50">
-                          <td className="py-3 font-medium text-foreground">{g.skill}</td>
-                          <td className="py-3 text-muted-foreground">{g.have}</td>
-                          <td className="py-3 text-muted-foreground">{g.need}</td>
-                          <td className={`py-3 font-medium ${gapColor[g.gapLevel]}`}>{g.gapLevel}</td>
-                        </tr>
-                      ))}
+                      {activeSession.gapAnalysis.map((g, idx) => {
+                        const isExpanded = !!expandedGaps[idx];
+                        return (
+                          <React.Fragment key={idx}>
+                            <tr
+                              onClick={() => toggleGap(idx)}
+                              className="border-b border-border/50 hover:bg-secondary/20 transition-colors group cursor-pointer"
+                            >
+                              <td className="py-4 pl-4 font-medium text-foreground">
+                                <div className="flex items-center gap-2">
+                                  <ChevronDown className={`w-4 h-4 transition-transform duration-200 ${isExpanded ? "rotate-180" : ""}`} />
+                                  {g.skill}
+                                </div>
+                              </td>
+                              <td className="py-4 text-muted-foreground">{g.have}</td>
+                              <td className="py-4 text-muted-foreground">{g.need}</td>
+                              <td className={`py-4 font-medium ${gapColor[g.gapLevel]}`}>{g.gapLevel}</td>
+                              <td className="py-4 pr-4"></td>
+                            </tr>
+                            {isExpanded && (
+                              <tr>
+                                <td colSpan={5} className="p-0">
+                                  <div className="bg-secondary/30 p-4 rounded-b-xl border-x border-b border-border/50 mx-2 mb-2">
+                                    <h4 className="text-xs font-semibold text-foreground uppercase tracking-wider mb-3">Recommended Learning Resources</h4>
+                                    <div className="flex flex-wrap gap-3">
+                                      {g.resources && g.resources.length > 0 ? (
+                                        g.resources.map((resKey, rIdx) => {
+                                          const resource = RESOURCE_MAPPING[resKey];
+                                          if (resource) {
+                                            return (
+                                              <a
+                                                key={rIdx}
+                                                href={resource.url}
+                                                target="_blank"
+                                                rel="noopener noreferrer"
+                                                className="inline-flex items-center gap-1.5 px-3 py-1.5 rounded-lg bg-background border border-border hover:border-primary/50 text-sm text-primary transition-all shadow-sm"
+                                              >
+                                                <BookOpen className="w-3.5 h-3.5" />
+                                                {resource.label}
+                                              </a>
+                                            );
+                                          }
+                                          return (
+                                            <a
+                                              key={rIdx}
+                                              href={`https://www.google.com/search?q=${encodeURIComponent(resKey + " documentation")}`}
+                                              target="_blank"
+                                              rel="noopener noreferrer"
+                                              className="inline-flex items-center gap-1.5 px-3 py-1.5 rounded-lg bg-background border border-border hover:border-primary/50 text-sm text-primary transition-all shadow-sm"
+                                            >
+                                              <Search className="w-3.5 h-3.5" />
+                                              {resKey}
+                                            </a>
+                                          );
+                                        })
+                                      ) : (
+                                        <p className="text-xs text-muted-foreground italic">No specific resources found. Try searching for "{g.skill} docs".</p>
+                                      )}
+                                    </div>
+                                  </div>
+                                </td>
+                              </tr>
+                            )}
+                          </React.Fragment>
+                        );
+                      })}
                     </tbody>
                   </table>
                 </div>

--- a/src/pages/JobTrackerPage.tsx
+++ b/src/pages/JobTrackerPage.tsx
@@ -16,6 +16,8 @@ import {
   ChevronRight,
   Search,
   X,
+  Check,
+  ChevronsUpDown,
 } from "lucide-react";
 import { Button } from "@/components/ui/button";
 import { Input } from "@/components/ui/input";
@@ -372,9 +374,23 @@ export default function JobTrackerPage({ jobs, sessions, onAddJob, onUpdateJob, 
   const [selectedDate, setSelectedDate] = useState<string | null>(null);
   const [showCalendarModal, setShowCalendarModal] = useState(false);
   const [form, setForm] = useState({ companyName: "", jobTitle: "", jobUrl: "", status: "Applied" as Status });
+  const [searchQuery, setSearchQuery] = useState("");
+  const [statusFilter, setStatusFilter] = useState("all");
+
   const { toast } = useToast();
   const navigate = useNavigate();
   const location = useLocation();
+
+  const filteredJobs = localJobs.filter((job) => {
+    const matchesSearch =
+      job.companyName.toLowerCase().includes(searchQuery.toLowerCase()) ||
+      job.jobTitle.toLowerCase().includes(searchQuery.toLowerCase()) ||
+      (job.location && job.location.toLowerCase().includes(searchQuery.toLowerCase()));
+    
+    const matchesStatus = statusFilter === "all" || job.status === statusFilter;
+    
+    return matchesSearch && matchesStatus;
+  });
 
   // Handle route state for auto-selection
   useEffect(() => {
@@ -675,6 +691,7 @@ export default function JobTrackerPage({ jobs, sessions, onAddJob, onUpdateJob, 
             </p>
           </div>
 
+<<<<<<< HEAD
           <div className="flex items-center gap-2 flex-wrap">
             {/* Search bar (issue #188) */}
             <div className="relative">
@@ -717,9 +734,38 @@ export default function JobTrackerPage({ jobs, sessions, onAddJob, onUpdateJob, 
             )}
             <div className="flex bg-secondary rounded-lg p-0.5">
               <button onClick={() => setView("kanban")} className={`px-3 py-1.5 rounded-md text-sm transition-colors ${view === "kanban" ? "bg-card text-foreground shadow-sm" : "text-muted-foreground"}`}>
+=======
+          <div className="flex items-center gap-3 flex-wrap">
+            <div className="relative w-48 sm:w-64">
+              <Search className="absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground" />
+              <Input
+                type="text"
+                placeholder="Search jobs or companies..."
+                value={searchQuery}
+                onChange={(e) => setSearchQuery(e.target.value)}
+                className="pl-9 h-9 bg-secondary/50 border-border"
+              />
+            </div>
+            <Select value={statusFilter} onValueChange={setStatusFilter}>
+              <SelectTrigger className="w-32 h-9 bg-secondary/50 border-border">
+                <SelectValue placeholder="All Statuses" />
+              </SelectTrigger>
+              <SelectContent>
+                <SelectItem value="all">All Statuses</SelectItem>
+                {STATUSES.map((s) => (
+                  <SelectItem key={s} value={s}>
+                    {s}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+
+            <div className="flex bg-secondary rounded-lg p-0.5 h-9 items-center">
+              <button onClick={() => setView("kanban")} className={`px-3 py-1.5 rounded-md text-sm transition-colors h-8 ${view === "kanban" ? "bg-card text-foreground shadow-sm" : "text-muted-foreground"}`}>
+>>>>>>> e9fc0f7 (feat: add search & filter to Job Tracker and fresher onboarding toggle)
                 <LayoutGrid className="w-4 h-4" />
               </button>
-              <button onClick={() => setView("table")} className={`px-3 py-1.5 rounded-md text-sm transition-colors ${view === "table" ? "bg-card text-foreground shadow-sm" : "text-muted-foreground"}`}>
+              <button onClick={() => setView("table")} className={`px-3 py-1.5 rounded-md text-sm transition-colors h-8 ${view === "table" ? "bg-card text-foreground shadow-sm" : "text-muted-foreground"}`}>
                 <TableIcon className="w-4 h-4" />
               </button>
             </div>
@@ -1090,7 +1136,11 @@ export default function JobTrackerPage({ jobs, sessions, onAddJob, onUpdateJob, 
                 {filteredJobs.length === 0 ? (
                   <tr>
                     <td colSpan={5} className="py-12 text-center text-muted-foreground">
+<<<<<<< HEAD
                       {hasActiveFilter ? "No applications match your filters" : "No applications yet"}
+=======
+                      {localJobs.length === 0 ? "No applications yet" : "No matching applications found"}
+>>>>>>> e9fc0f7 (feat: add search & filter to Job Tracker and fresher onboarding toggle)
                     </td>
                   </tr>
                 ) : (

--- a/src/pages/OnboardingPage.tsx
+++ b/src/pages/OnboardingPage.tsx
@@ -183,6 +183,7 @@ export default function OnboardingPage({ user, profile, onSave }: OnboardingPage
   const [softSkills, setSoftSkills] = useState<string[]>([]);
   const [fears, setFears] = useState<string[]>([]);
   const [fearNotes, setFearNotes] = useState("");
+  const [isFresher, setIsFresher] = useState(false);
 
   useEffect(() => {
     if (!profile) {
@@ -196,7 +197,9 @@ export default function OnboardingPage({ user, profile, onSave }: OnboardingPage
     setGraduationYear(profile.graduationYear);
     setCoursework(profile.coursework);
     setCertifications(profile.certifications);
-    setWorkHistory(profile.workHistory.length > 0 ? profile.workHistory : [emptyWorkEntry()]);
+    const fresher = profile.workHistory.length === 0;
+    setIsFresher(fresher);
+    setWorkHistory(fresher ? [] : (profile.workHistory.length > 0 ? profile.workHistory : [emptyWorkEntry()]));
     setTechnicalSkills(profile.technicalSkills);
     setSoftSkills(profile.softSkills);
     setFears(profile.interviewFears);
@@ -240,8 +243,12 @@ export default function OnboardingPage({ user, profile, onSave }: OnboardingPage
         return null;
       }
       case 2: {
+<<<<<<< HEAD
         if (isFresher) return null; // freshers skip work validation
 
+=======
+        if (isFresher) return null;
+>>>>>>> e9fc0f7 (feat: add search & filter to Job Tracker and fresher onboarding toggle)
         let filledCount = 0;
         const currentYear = new Date().getFullYear();
 
@@ -430,6 +437,7 @@ export default function OnboardingPage({ user, profile, onSave }: OnboardingPage
 
               {step === 2 && (
                 <>
+<<<<<<< HEAD
                   {/* Fresher toggle (issue #61) */}
                   <div className="flex items-start gap-3 p-3 rounded-xl bg-primary/5 border border-primary/20">
                     <Checkbox
@@ -451,6 +459,25 @@ export default function OnboardingPage({ user, profile, onSave }: OnboardingPage
                     </div>
                   </div>
 
+=======
+                  <div className="flex items-center gap-2 mb-4 bg-secondary/20 p-3 rounded-xl border border-border/50">
+                    <Checkbox
+                      id="fresher-toggle"
+                      checked={isFresher}
+                      onCheckedChange={(checked) => {
+                        setIsFresher(!!checked);
+                        if (checked) {
+                          setWorkHistory([]);
+                        } else {
+                          setWorkHistory([emptyWorkEntry()]);
+                        }
+                      }}
+                    />
+                    <Label htmlFor="fresher-toggle" className="cursor-pointer text-sm font-medium text-foreground">
+                      I am a fresher (No professional work experience)
+                    </Label>
+                  </div>
+>>>>>>> e9fc0f7 (feat: add search & filter to Job Tracker and fresher onboarding toggle)
                   {!isFresher && (
                     <>
                       {workHistory.map((entry, idx) => (

